content/gocourse-2019-05-melbourne.slide

```diff
@@ -1003,6 +1003,46 @@ Use `gofmt` or `goimport` as part of your editor/IDE setup
 - ...
 
 
+* JSON - Marshal / Unmarshal
+
+- Encoding a struct or primitive converts it into a json encoded stream
+- Marshalling uses an encoder to convert a struct / primitive into an array of json bytes, which are stored in memory
+- Decoding / Unmarshalling are the opposing functions to encoding and marshalling
+
+	func Marshal(v interface{}) ([]byte, error) {
+
+	}
+
+	func Unmarshal(data []byte, v interface{}) error {
+
+	}
+
+
+* JSON - Tags
+
+- `json` tags can be attached to struct fields to control json encoding 
```

Comment:
```
 attach `json` tag to specify JSON property name
```
I'd also add that a common error is to Go fields name private (lower case) which makes them not be (un) marshalled.